types/execution/execution.interfaces.ts

```diff
@@ -66,7 +66,8 @@ export type ConfiguredRuntimeTool = {
 export type ConfiguredRuntimeTools = ConfiguredRuntimeTool[];
 
 export type ExecutableExecutionOptions = {
-    args: string[];
+    // TODO: narrow to string[], after clientstate is properly TSified
+    args: string | string[];
```

Comment:
oh you're changing it... but this isn't ever supposed to be just a string.. where is it used that way? can we get rid of that instead?

Reply:
right, but I explicitly made ExecutableExecutionOptions to get rid of the `string`
in the cmake call it's aparantly already fixed with https://github.com/compiler-explorer/compiler-explorer/blob/601c3ec3d70230adb53873185b56d14bb52c3456/lib/base-compiler.ts#L2555 - that is imo a preferable solution, because we "know" it's actually always `string[]` at that point because of this line https://github.com/compiler-explorer/compiler-explorer/blob/601c3ec3d70230adb53873185b56d14bb52c3456/lib/handlers/compile.ts#L460

Reply:
So as soon as the ExecutableExecutionOptions come into the backend, they shouldn't be actually `string` anymore.

But then I read this and noticed there's a bug in this code: https://github.com/compiler-explorer/compiler-explorer/blob/601c3ec3d70230adb53873185b56d14bb52c3456/lib/handlers/compile.ts#L441

Should be ([] instead of '')
```
args: Array.isArray(execReqParams.args)
                ? execReqParams.args || []
                : utils.splitArguments(execReqParams.args),
```

In the `cmake` call we also do this case https://github.com/compiler-explorer/compiler-explorer/blob/601c3ec3d70230adb53873185b56d14bb52c3456/lib/base-compiler.ts#L2555 which could work for the conversion to ExecutableExecutionOptions where needed

Reply:
Thanks for this! I'll go back to this and insist more on this type. Sometime in the near days.